tests/pfcwd/test_pfcwd_function.py

```diff
@@ -0,0 +1,518 @@
+
+        time.sleep(5)
+
+        # storm detect
+        logger.info("Verify if PFC storm is detected on port {}".format(port))
+        loganalyzer.analyze(marker)
+        self.stats.get_pkt_cnts(self.queue_oid, begin=True)
+        # test pfcwd functionality on a storm
+        self.traffic_inst.verify_wd_func(action if action != "dontcare" else "drop")
+        return loganalyzer
+
+    def storm_restore_path(self, loganalyzer, port, action):
+        """
+        Storm restoration action and associated verifications
+
+        Args:
+            loganalyzer(Loganalyzer) : loganalyzer instance
+            port(string) : DUT port
+            action(string) : PTF test action
+        """
+        marker = loganalyzer.update_marker_prefix("pfc_function_storm_restore_{}_port_{}".format(action, port))
+        ignore_file = os.path.join(TEMPLATES_DIR, "ignore_pfc_wd_messages")
+        reg_exp = loganalyzer.parse_regexp_file(src=ignore_file)
+        loganalyzer.ignore_regex.extend(reg_exp)
+        loganalyzer.expect_regex = []
+        loganalyzer.expect_regex.extend([EXPECT_PFC_WD_RESTORE_RE])
+        loganalyzer.match_regex = []
+
+        self.storm_hndle.stop_storm()
+        time.sleep(self.timers['pfc_wd_wait_for_restore_time'])
+        # storm restore
+        logger.info("Verify if PFC storm is restored on port {}".format(port))
+        loganalyzer.analyze(marker)
+        self.stats.get_pkt_cnts(self.queue_oid, begin=False)
+
+    def run_test(self, dut, port, action):
+        """
+        Test method that invokes the storm detection and restoration path which includes the traffic
+        test and associated counter verifications
+
+        Args:
+            dut(AnsibleHost) : DUT instance
+            port(string) : DUT port
+            action(string) : PTF test action
+        """
+        logger.info("--- Storm detection path for port {} ---".format(port))
+        loganalyzer = self.storm_detect_path(dut, port, action)
+        logger.info("--- Storm restoration path for port {} ---".format(port))
+        self.storm_restore_path(loganalyzer, port, action)
+        logger.info("--- Verify PFCwd counters for port {} ---".format(port))
+        self.stats.verify_pkt_cnts(self.pfc_wd['port_type'], self.pfc_wd['test_pkt_count'])
+
+    def test_pfcwd_actions(self, request, setup_pfc_test, fanout_graph_facts, ptfhost, duthost, fanouthosts):
+        """
+        PFCwd functional test
+
+        Args:
+            request(object) : pytest request object
+            setup_pfc_test(fixture) : Module scoped autouse fixture for PFCwd
+            fanout_graph_facts(fixture) : fanout graph info
+            ptfhost(AnsibleHost) : ptf host instance
+            duthost(AnsibleHost) : DUT instance
+            fanouthosts(AnsibleHost): fanout instance
+        """
+        setup_info = setup_pfc_test
+        self.fanout_info = fanout_graph_facts
+        self.ptf = ptfhost
```

Comment:
Just curios to know if the fanout_info is added to the class object. My experience with those test classes is that I could not add members to them.

Reply:
yes